### DISCUSSION TITLE
Application for Alaska Wildfire Prediction Using Satellite Imagery - GSOC

### DISCUSSION THREAD

**Discussion by neelpatel8798** [YaliWang2019/AK-Satellite-Imagery-Wildfire-Prediction#10] [General]
Hello @YaliWang2019, 

My name is Neel Patel, and I am currently pursuing a Master’s in Computer Science at the University of Maryland Baltimore County (UMBC). Currently, I'm working under Dr. Tejas Gokhale (UMBC), on Deep Learning and Computer Vision algorithms for geospatial analytics. The project aims are to develop a method to train neural networks to understand agricultural crop indices from satellite images (HLS, Sentinel-2 and MODIS dataset), tracking the evolution of those indices over time, and reasoning about the factors that may have caused those changes, including human and/or natural phenomena. This algorithm will leverage large language models (LLMs) to convert the predictions made by the neural network model into natural language explanations and insights that can be understood by common users. 

One of the main contributions to my research project is that I have created a pipeline that downloads HLS data and CDL according to the deep learning model input (224 × 224, 256 × 256, 512 × 512 etc.) across the United States. The pipeline is created in such a way that it supports various regions such as County, State, NASS Region, Entire US and Custom Shape-file across the United States covering (2013 to present for Landsat 8/9 data and 2015 to present for Sentinel-2A/B data). This pipeline is designed and created in 2 parts, a) USDA (Cropland Data Layer) CDL Layer download based on the crop type as a ground truth data. b) Original HLS (Landsat and Sentinel Data) based on the crop type and selected region across CONUS. In the part a) The pipeline supports more than 100 CDL crop types across the contiguous US at 30-meter resolution, with high accuracy for major crops like corn and soybeans. In the Second part b) Raw HLS data is downloaded based on filters such as Cloud Cover, Spatial Cover, QA Band Assessment. The final output of this pipeline includes CDL layer data (ground truth) and processed HLS data (model input), as 6-Band images, computed NDVI, EVI, colored image and false color image.  

Before starting my master's study, I worked at TCS Research and Innovation, as a Senior Python Developer for 3 years. My responsibilities include delivering different POCs and the development/implementation of various use cases developed by the research team into Cloud Platform architecture. One of my works was to develop Cloud Native Data Publish Pipeline POCs, where I developed solutions for agriculture-related to geo-spatial use-cases for detecting various phenological conditions and health conditions of the crops into productization that is deployed on Amazon Web Services using Python, Machine Learning, Deep Learning, and Remote Sensing techniques and automated the process of publish and unpublish pipeline. This work was admired by AWS, they featured our use-case in the [AWS Sustainability Partner Innovator Stories](https://aws.amazon.com/blogs/apn/tcs-digital-farming-generates-near-real-time-insights-on-crop-conditions-for-actionable-decision-making/) and published TCS's Digital Farming Video on [This is My Architecture Series](https://www.youtube.com/watch?v=Dxq_U1TNx1s). 

**Project Plan:**
As the Outcome of this Research project is  
1) Data Processing and Downloading Pipeline, which automatically downloads and processes different combinations of Satellite Image and ground-based weather data. (Covering all preprocessing task such as band selection, indices calculation, geospatial cropping, cloud covering, spatial filtering, band QA assessment) 

2) For model training and evaluation, we can use various model architectures for prediction of the wildfire across Alaska, as well as segmentation tasks to understand particular regions of wildfire across time-series data. This will help us to understand the pattern of fire spreading prediction. Due to the availability of multi-modal data, we can also implement Domain adaptation and Domain Generalization tasks, this will help us to understand the domain shift across time, region or dataset related to wildfire. Along with the visualization of the data using Web-Based dashboard (using Geoserver and postGIS database) which includes, statistics, dataset selection, change analysis, before and after affects of wildfire in particular region etc. 

3) Final  Report based on the model's performance, pipeline data downloading and processing pipeline (which can be extended to other countries.), This work we can also contribute as a potential Research Paper. 

Technologies:
Languages and Tools: Python, Github, Git, HTML, CSS
Framework: PyTorch, Keras, OpenCV, Flask/Django (UI visualization)
Database: For raster, local storage or any cloud storage. For vector PostGIS/PostgreSQL
Geospatial: QGIS, ArcGIS, Rasterio, GDAL, Geoserver 
Cloud Service (optional): AWS (Geospatial ML with Amazon SageMaker). 
Dataset Source: a) Landsat 8&9 (AWS Open Registry (Paid), NASA EarthData Search) 
                            b) Sentinel-1&2 (AWS Open Registry (Free))
                            c) Modis (NASA EarthData-LPDAAC)
                            d) VIIRS (NASA EarthData-OPDAAC)
                            e) VIIRS (NASA EarthData-OPDAAC)
                            f) ALOS-2 (SAT Platform - JAXA)
                            g) Not sure about ground data source. 

Looking forward to discussing more in detail about my experience and project. 

--
Thanks & Regards,
Neel R Patel,
MS Computer Science @UMBC | Prev @TCS Research and Innovation
M: +1 443-682-3348 | Whatsapp: +91 70418 57439
Email: n.digiid@gmail.com
